framework/db/Query.php

```diff
@@ -590,45 +590,51 @@ public function addSelect($columns)
         if ($this->select === null) {
             $this->select = $columns;
         } else {
-            $this->select = array_merge($this->select, $this->removeDuplicatedColumns($columns));
+            $this->select = array_merge($this->select, $this->getUniqueColumns($columns));
         }
 
         return $this;
     }
 
     /**
-     * Removes duplicated columns from given columns definitions.
+     * Returns unique column names excluding duplicates.
      * Columns to be removed:
      * - if column definition already present in SELECT part with same alias
      * - if column definition without alias already present in SELECT part without alias too
      * @param array $columns the columns to be merged to the select.
+     * @since 2.0.14
      */
-    protected function removeDuplicatedColumns($columns)
+    protected function getUniqueColumns($columns)
     {
-        $simple_columns = $this->filterSimpleColumnsFromSelect();
+        $simpleColumns = $this->getUnaliasedColumnsFromSelect();
+        $columns = array_unique($columns);
 
-        foreach ($columns as $columnName => $columnDefinition) {
-            if ($columnDefinition instanceof Query)
+        foreach ($columns as $columnAlias => $columnDefinition) {
+            if ($columnDefinition instanceof Query) {
                 continue;
+            }
 
-            if ((is_string($columnName) && isset($this->select[$columnName]) && $this->select[$columnName] === $columnDefinition) ||
-                (is_integer($columnName) && in_array($columnDefinition, $simple_columns)))
-                unset($columns[$columnName]);
+            if ((is_string($columnAlias) && isset($this->select[$columnAlias]) && $this->select[$columnAlias] === $columnDefinition) ||
+                (is_integer($columnAlias) && in_array($columnDefinition, $simpleColumns))) {
```

Comment:
in_array is still there and array_unique was added. Doesn't make sense to me.

Reply:
I've removed `in_array` in `getUnaliasedColumnsFromSelect()`. This `in_array` is inteded to check if an unalised column is already present.